### PR TITLE
Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks

### PR DESCRIPTION
**Pull request by luojiyin1987**
[PR-77](https://github.dev/FreeCodeCamp-Chengdu/Wiki/pull/77) [PR-77](https://codespaces.new/FreeCodeCamp-Chengdu/Wiki/pull/77) [PR-77](https://gitpod.io/?autostart=true#https://github.com/FreeCodeCamp-Chengdu/Wiki/pull/77) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=FreeCodeCamp-Chengdu&utm_campaign=badge_info)

@FreeCodeCamp-Chengdu I create a Pull Request to close #issue_number

### CODE REVIEW

**Review by luojiyin1987 — COMMENTED** [review]
fix

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    - https
+    - 内网
+    - SSL
+    - caddy
+    - DevOps
+    - Docker
+toc: true
+---
+
+# 使用 Caddy 和 CloudFlare 在内网自动签发 https 证书
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:19]:
```suggestion
    - Caddy
    - DevOps
    - Docker
toc: true
---
```

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+## 前置条件
+
+1.  Caddy 和 dns-cloudflare 插件，需要构建自己的 Docker 镜像
+2.  域名的 NS 服务器设置为 cloudflare， 配置好，大概一天后生效（保守估计）
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:24]:
```suggestion
1.  Caddy 和 `dns-cloudflare` 插件，需要构建自己的 Docker 镜像
2.  域名的 NS 服务器设置为 CloudFlare， 配置好，大概一天后生效（保守估计）
```

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+3.  内网 DNS 服务器， 如果路由器支持定义局域网域名，可以不安装，我用的是 adguard
+4.  内网 Ubuntu 服务器
+5.  Ubuntu 安装 Docker
+
+## 构建 Docker 镜像
+
+在 docker compose 同一层 文件夹里，创建一个 Dockerfile
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:31]:
```suggestion
在 Docker compose 同一层文件夹里，创建一个 Dockerfile：
```

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+```yaml
+FROM  caddy:builder AS builder
+
+RUN caddy-builder \
+github.com/caddy-dns/cloudflare
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:37]:
```suggestion
```Dockerfile
FROM  caddy:builder AS builder

RUN caddy-builder github.com/caddy-dns/cloudflare
```

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+## 获取用户 API 令牌
+
+-   登录到 Cloudflare Dashboard 。
+-   前往 "My Profile" > "[API Tokens](https://dash.cloudflare.com/profile/api-tokens)" 。
+-   确认你的 API 令牌具有所需的权限（Zone.Zone:Read 和 Zone.DNS:Edit），并且选择要使用的域名。
+    在 docker compose 同一层 文件夹里 创建一个 `.env` 文件
+
+```yaml
+CLOUDFLARE_API_TOKEN = your token
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:65]:
```suggestion
```ini
CLOUDFLARE_API_TOKEN = your token
```

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+在 docker compose 同一层 文件夹里，创建一个 Caddyfile 文件
+
+```yaml
+caddy.luojiyin.top {
+  tls {
+       dns cloudflare {env.CLOUDFLARE_API_TOKEN}
+  }
+  respond / "Hello, world! This is a static response from Caddy." 200
+}
+```
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:85]: 语法名不对

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+            # It is important to have these to conserve data after the container restarts
+            - "./data:/data"
+            # Unless we want to SSH to our server every time it restarts.
+            - "./config:/config"
+            - "./Caddyfile:/etc/caddy/Caddyfile"
+        restart: unless-stopped
+
+networks:
+    caddy:
+        name: caddy
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:121]: 网络是依赖项（且内容少），声明在上方。

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+完整的文件夹目录
+
+```shell
+├── Caddyfile
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:127]:
```suggestion
```text
├── Caddyfile
```

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+没有报错信息， 基本成功了。
+
+## 配置内网 DNS
+
+<figure>
+<img
+src="Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks/e280d22c8bd57a59b01707b2b37e02156862500a.png"
+title="wikilink">
+<figcaption
+aria-hidden="true">配置内网域名解析</figcaption>
```
Comment (TechQuery) [inline _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:166]:
```suggestion
<figcaption>配置内网域名解析</figcaption>
```

**Review by TechQuery — CHANGES_REQUESTED** [review]

File: _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md (hunk 6 above)
Comment (luojiyin1987) [reply _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md:85]: 看不出来

Comment (TechQuery) [reply _posts/Article/Engineering/Using-Caddy-and-cloudflare-to-issue-wildcards-on-private-networks.md]: > 看不出来

肯定不是 YAML，先写个 `caddyfile` 的语言名吧，后面哪个 Markdown 解析器支持就直接能高亮。

**Review by TechQuery — APPROVED** [review]

**Pull request merged**